google-cloud-channel-v1/AUTHENTICATION.md

```diff
@@ -0,0 +1,169 @@
+# Authentication
+
+In general, the google-cloud-channel-v1 library uses
+[Service Account](https://cloud.google.com/iam/docs/creating-managing-service-accounts)
+credentials to connect to Google Cloud services. When running within
+[Google Cloud Platform environments](#google-cloud-platform-environments) the
+credentials will be discovered automatically. When running on other
+environments, the Service Account credentials can be specified by providing the
+path to the
+[JSON keyfile](https://cloud.google.com/iam/docs/managing-service-account-keys)
+for the account (or the JSON itself) in
+[environment variables](#environment-variables). Additionally, Cloud SDK
+credentials can also be discovered automatically, but this is only recommended
+during development.
+
+## Quickstart
+
+1. [Create a service account and credentials](#creating-a-service-account).
+2. Set the [environment variable](#environment-variables).
+
+```sh
+export CHANNEL_CREDENTIALS=path/to/keyfile.json
+```
+
+3. Initialize the client.
+
+```ruby
+require "google/cloud/channel/v1"
+
+client = ::Google::Cloud::Channel::V1::CloudChannelService::Client.new
+```
+
+## Credential Lookup
+
+The google-cloud-channel-v1 library aims to make authentication
+as simple as possible, and provides several mechanisms to configure your system
+without requiring **Service Account Credentials** directly in code.
+
+**Credentials** are discovered in the following order:
+
+1. Specify credentials in method arguments
+2. Specify credentials in configuration
+3. Discover credentials path in environment variables
+4. Discover credentials JSON in environment variables
+5. Discover credentials file in the Cloud SDK's path
+6. Discover GCP credentials
+
+### Google Cloud Platform environments
+
+When running on Google Cloud Platform (GCP), including Google Compute Engine
+(GCE), Google Kubernetes Engine (GKE), Google App Engine (GAE), Google Cloud
+Functions (GCF) and Cloud Run, **Credentials** are discovered automatically.
+Code should be written as if already authenticated.
+
+### Environment Variables
+
+The **Credentials JSON** can be placed in environment variables instead of
+declaring them directly in code. Each service has its own environment variable,
+allowing for different service accounts to be used for different services. (See
+the READMEs for the individual service gems for details.) The path to the
+**Credentials JSON** file can be stored in the environment variable, or the
+**Credentials JSON** itself can be stored for environments such as Docker
+containers where writing files is difficult or not encouraged.
+
+The environment variables that google-cloud-channel-v1
+checks for credentials are configured on the service Credentials class (such as
+{::Google::Cloud::Channel::V1::CloudChannelService::Credentials}):
+
+1. `CHANNEL_CREDENTIALS` - Path to JSON file, or JSON contents
```

Comment:
Since `Channel` is an abstraction in the client auth domain, is this env var name too generic and at risk of collision with someone's existing `CHANNEL_CREDENTIALS` env var? I'm not convinced of this, just offering a thought.

Reply:
That's a tough one. I share your discomfort, but I'm also not convinced it's worth making an exception to our normal naming practice. I think my inclination is to leave it.

Reply:
SGTM